File: day1/contains_duplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        # check_set = set()

        # for n in nums:
        #     if n not in check_set:
        #         check_set.add(n)
        #     else:
        #         return True
        # return False

        return len(set(nums)) != len(nums)

File: day1/contains_duplicate.cpp
#include<iostream>
#include<vector>
#include<map>


class Solution {
public:
    bool containsDuplicate(std::vector<int>& nums) {
        std::map<int, int> valueCount;
        bool result = false;
        for(int i : nums){
            //checking if value is in the map already
            if(valueCount.find(i) != valueCount.end()){
                result = true;
                break;
            }
            else{
                ++valueCount[i]; 
            }
        }

        return result;         
    }
};